precompile/fee_config_manager.go

```diff
@@ -84,41 +84,44 @@ func PackGetFeeConfigInput() []byte {
 	return getFeeConfigSignature
 }
 
-// PackGetLastChangedAtInput packs the getLastChangedAt signature
+// PackGetLastChangedAtInput packs the getFeeConfigLastChangedAt signature
 func PackGetLastChangedAtInput() []byte {
-	return getLastChangedAtSignature
+	return getFeeConfigLastChangedAtSignature
 }
 
 // PackFeeConfig packs [feeConfig] without the selector into the appropriate arguments for fee config operations.
 func PackFeeConfig(feeConfig commontype.FeeConfig) ([]byte, error) {
 	//  input(feeConfig)
-	return packHelper(feeConfig, false)
+	return packFeeConfigHelper(feeConfig, false), nil
 }
 
 // PackSetFeeConfig packs [feeConfig] with the selector into the appropriate arguments for setting fee config operations.
 func PackSetFeeConfig(feeConfig commontype.FeeConfig) ([]byte, error) {
 	// function selector (4 bytes) + input(feeConfig)
-	return packHelper(feeConfig, true)
+	return packFeeConfigHelper(feeConfig, true), nil
 }
 
-func packHelper(feeConfig commontype.FeeConfig, useSelector bool) ([]byte, error) {
-	fullLen := feeConfigInputLen
-	packed := [][]byte{
-		feeConfig.GasLimit.FillBytes(make([]byte, common.HashLength)),
-		new(big.Int).SetUint64(feeConfig.TargetBlockRate).FillBytes(make([]byte, common.HashLength)),
-		feeConfig.MinBaseFee.FillBytes(make([]byte, common.HashLength)),
-		feeConfig.TargetGas.FillBytes(make([]byte, common.HashLength)),
-		feeConfig.BaseFeeChangeDenominator.FillBytes(make([]byte, common.HashLength)),
-		feeConfig.MinBlockGasCost.FillBytes(make([]byte, common.HashLength)),
-		feeConfig.MaxBlockGasCost.FillBytes(make([]byte, common.HashLength)),
-		feeConfig.BlockGasCostStep.FillBytes(make([]byte, common.HashLength)),
+func packFeeConfigHelper(feeConfig commontype.FeeConfig, useSelector bool) []byte {
+	hashes := []common.Hash{
+		common.BigToHash(feeConfig.GasLimit),
+		common.BigToHash(new(big.Int).SetUint64(feeConfig.TargetBlockRate)),
+		common.BigToHash(feeConfig.MinBaseFee),
+		common.BigToHash(feeConfig.TargetGas),
+		common.BigToHash(feeConfig.BaseFeeChangeDenominator),
+		common.BigToHash(feeConfig.MinBlockGasCost),
+		common.BigToHash(feeConfig.MaxBlockGasCost),
+		common.BigToHash(feeConfig.BlockGasCostStep),
 	}
+
 	if useSelector {
-		packed = append([][]byte{setFeeConfigSignature}, packed...)
-		return packOrderedHashesWithSelector(packed, fullLen+selectorLen)
+		res := make([]byte, len(setFeeConfigSignature)+len(hashes)*common.HashLength)
```

Comment:
isn't  `len(hashes)*common.HashLength)` equivalent to `feeConfigInputLen`?

Reply:
Yes, updated.